### ISSUE TITLE
[Bug][Ripples] New Stencil functions in LÖVE 0.10.X

### ISSUE THREAD

**Issue by Ranguna** [Positive07/material-love#20]
Since 0.10.0, the stencil functions have been changed in horrible, horrible ways that make no sence, love.graphics.setStencil has dissapeared and it has been replace by [two](https://love2d.org/wiki/love.graphics.stencil) [other](https://love2d.org/wiki/love.graphics.setStencilTest) terrible functions.
Ripples [use the old stencil function](https://github.com/Positive07/material-love/blob/master/libs/ripple.lua#L97), this will need to be changed once you release 2.0.0
Best of luck understanding the new stencil API.

**Comment by pablomayobre** [comment]
Yes I know, this library was not made for 0.10.0 (Since I can't run that in my PC) but I intent to fix this in 2.0.0 whenever that is released.

I may fix this out of boredom, but don't know when. I accept pull requests too

Thanks for the support! I'm glad there is still some interest in this

**Comment by TangentFoxy** [comment]
Since I just stumbled across this and would like to figure out how to fix this and/or make it compatible with both versions, can either of you give me a few pointers to how to get started figuring it out?

**Comment by pablomayobre** [comment]
Okey I went on and fixed it since I saw interest in this. As I explained in the description of the commit, the new `love.graphic.stencil` function takes a first argument which is a function, it draws this function to a buffer (think a canvas) with the mode specified as second argument which is a StencilAction, in this case we used replace, this means that every time we draw to a pixel that pixel will get the new value we set, in this case we specified with the third argument that each pixel will be set to 1, then there is an additional value that can help to combine stencil functions, but we dont care about it.

Then we need to set the test, this is done with `love.graphics.setTencilTest` this function will make LÖVE check if the pixel we are drawing at passes the test, to do this it receives a first argument which in this case is `"greater"` to specified that to pass the test that pixel of the stencil buffer needs to have a value greater than the value specified as second argument of this function, in this case 0.

That's the gist of it, the fix is quite silly since we use pretty much all default values. Thanks for the support, if you find any issue feel free to report, I'll fix them one by one, since I can't run 0.10.0 on my PC
